platform/darwin/src/MGLOpenGLStyleLayer.mm

```diff
@@ -107,16 +107,6 @@ - (instancetype)initWithIdentifier:(NSString *)identifier {
     return (mbgl::style::CustomLayer *)super.rawLayer;
 }
 
-#if TARGET_OS_IPHONE
-- (EAGLContext *)context {
-    return self.style.mapView.context;
-}
-#else
-- (CGLContextObj)context {
-    return self.style.mapView.context;
```

Comment:
This property is still declared in the public header:

https://github.com/mapbox/mapbox-gl-native/blob/d8e5c5afcb8c0184012380e651881db457792af7/platform/darwin/src/MGLOpenGLStyleLayer.h#L30-L37

But calling this public method will crash without an implementation.

The property was added in #13978 to facilitate integrating custom layers with OpenGL-compatible graphics libraries like SceneKit (which is also Metal-compatible). SCNRenderer needs either an MTLDevice (for rendering to a Metal surface) or an EAGLContext or CGLContextObj (for rendering to OpenGL).

If at all possible, we should keep this property or move it to MGLMapView, but make it `nullable` so we can return `nil` under the right circumstances. An MTLDevice-typed property would be nice as well. 😉